src/web/demo.rkt

```diff
@@ -240,143 +240,94 @@
-                                                (test-context test)))))
-                      (get-output-string os)))))
-              (define derivations 
-                (for/list ([altn altns])
-                          (render-json altn
-                                                processed-pcontext
-                                                test-pcontext
-                                                (test-context test))))
-
-              (eprintf " complete\n")
-              (define hash-table (hasheq 'alternatives fpcores
-                      'histories histories
-                      'derivations derivations
-                      'splitpoints splitpoints))
-
-             (hash-set! *completed-jobs* hash hash-table)
-             (eprintf " complete\n")
-             (hash-remove! *jobs* hash)
-             (semaphore-post sema)]
-            [(list 'exacts hash formula sema seed sample)
-             (define test (parse-test formula))
-             (define pcontext (json->pcontext sample (test-context test)))
-             (eprintf "Ground truth job started on ~a..." formula)
-             (define result (run-herbie 'exacts test #:seed seed 
-              #:pcontext pcontext #:profile? #f #:timeline-disabled? #t))
-             (define exacts (job-result-backend result))
-             (hash-set! *completed-jobs* hash (hasheq 'points exacts))
-             (eprintf " complete\n")
-             (hash-remove! *jobs* hash)
-             (semaphore-post sema)]
-            [(list 'evaluate hash formula sema seed sample)
-             (define test (parse-test formula))
-             (define pcontext (json->pcontext sample (test-context test)))
-             (eprintf "Evaluation job started on ~a..." formula)
-             (define result (run-herbie 'evaluate test #:seed seed 
-              #:pcontext pcontext #:profile? #f #:timeline-disabled? #t))
-             (define approx (job-result-backend result))
-             (hash-set! *completed-jobs* hash (hasheq 'points approx))
-             (eprintf " complete\n")
-             (hash-remove! *jobs* hash)
-             (semaphore-post sema)]
-            [(list 'cost hash formula sema)
-             (define test (parse-test formula))
-             (eprintf "Computing cost of ~a..." formula)
-             (define result (run-herbie 'cost test 
-              #:profile? #f #:timeline-disabled? #t))
-             (define cost (job-result-backend result))
-             (hash-set! *completed-jobs* hash (hasheq 'cost cost))
-             (eprintf " complete\n")
-             (hash-remove! *jobs* hash)
-             (semaphore-post sema)])
+         [(list 'sample hash formula sema args)
+          (define seed (first args))
+          (define test (parse-test formula))
+          (eprintf "Sampling job started on ~a..." formula)
+          (define result (run-herbie 'sample test #:seed seed 
+          #:profile? #f #:timeline-disabled? #t))
```

Comment:
What's with the terrible indentation?

Reply:
Why disable the timeline? Does it matter? Same with profiling.

Reply:
timeline is disabled to have the same behavior as main. Though if all of these endpoints are on their own thread could turn timeline on? Or be an option sent over with the request?